src/main/java/spireMapOverhaul/zones/gremlinTown/patches/HordePatches.java

```diff
@@ -0,0 +1,200 @@
+                HordeHelper.onVictory();
+        }
+    }
+
+    @SpirePatch2(
+            clz = AbstractMonster.class,
+            method = "die",
+            paramtypez = boolean.class
+    )
+    public static class backAttackPatch {
+        @SpirePostfixPatch
+        public static void Postfix() {
+            if (AbstractDungeon.lastCombatMetricKey.equals(GremlinTown.GREMLIN_HORDE)) {
+                HordeHelper.calculateBackAttack();
+            }
+        }
+    }
+
+    @SpirePatch2(
+            clz = GainBlockRandomMonsterAction.class,
+            method = "update"
+    )
+    public static class shieldPatch {
+        @SpireInsertPatch (
+                locator = Locator.class,
+                localvars = {"validMonsters"}
+        )
+        public static void Insert(ArrayList<AbstractMonster> validMonsters) {
+            validMonsters.removeIf(AbstractCreature::isDeadOrEscaped);
+        }
+
+        public static class Locator extends SpireInsertLocator {
+            private Locator() {}
+
+            @Override
+            public int[] Locate(CtBehavior behavior) throws Exception {
+                Matcher matcher = new Matcher.MethodCallMatcher(ArrayList.class, "isEmpty");
+                return LineFinder.findInOrder(behavior, matcher);
+            }
+        }
+    }
+
+    @SpirePatch2(
+            clz = ApplyPowerAction.class,
+            method = "update"
+    )
+    public static class stackingBackAttackPatch {
+        @SpireInsertPatch (
+                locator = Locator.class
+        )
+        public static SpireReturn Insert(ApplyPowerAction __instance) {
+            AbstractPower pow = ReflectionHacks.getPrivate(__instance, ApplyPowerAction.class, "powerToApply");
+            AbstractCreature target = ReflectionHacks.getPrivate(__instance, AbstractGameAction.class, "target");
+            if (pow instanceof BackAttackPower && target.hasPower(BackAttackPower.POWER_ID)) {
+                __instance.isDone = true;
+                return SpireReturn.Return();
+            } else
+                return SpireReturn.Continue();
+        }
+
+        public static class Locator extends SpireInsertLocator {
+            private Locator() {}
+
+            @Override
+            public int[] Locate(CtBehavior behavior) throws Exception {
+                Matcher matcher = new Matcher.FieldAccessMatcher(ApplyPowerAction.class, "powerToApply");
+                return LineFinder.findInOrder(behavior, matcher);
+            }
+        }
+    }
+
+    @SpirePatch(
+            clz= AbstractMonster.class,
+            method=SpirePatch.CLASS
+    )
+    public static class GremlinField
+    {
+        public static SpireField<Boolean> horde = new SpireField<>(() -> false);
+    }
+
+    @SpirePatch2(
+            clz = AbstractMonster.class,
+            method = "applyPowers"
+    )
+    public static class StupidFuckingBackstabBullshitImSoMadDontChangeThisPatchNameGK {
```

Comment:
```suggestion
    public static class FixBackstabPatch {
```
DoN#t tell me what to do >:(